modules/tensor_mechanics/include/utils/RankTwoScalarTools.h

```diff
@@ -375,16 +377,96 @@ radialStress(const RankTwoTensorTempl<T> & stress,
-      radial_stress += radial_norm(j) * stress(j, i) * radial_norm(i);
+  T radial_stress = radial_norm * (stress * radial_norm);
 
   direction = radial_norm;
 
   return radial_stress;
 }
 
+/* The radial stress is calculated as
+ * radial_stress = radial^T_i * \sigma_{ij} * radial_j
+ * where normal is the position vector of the current point that is normal to
+ * the user-defined axis of rotation;
+ * @param center The center point of the spherical system
+ * @param curr_point The point corresponding to the stress (pass in & _q_point[_qp])
+ * @param direction The direction vector in which the scalar stress value is calculated.
+ */
+template <typename T>
+T
+radialStress(const RankTwoTensorTempl<T> & stress,
+             const Point & center,
+             const Point & curr_point,
+             Point & direction)
+{
+  Point radial = curr_point - center;
+  radial /= radial.norm();
+
+  // Compute the scalar stress component in the radial direction
+  T radial_stress = radial * (stress * radial);
+
+  direction = radial;
+
+  return radial_stress;
+}
+
+/* The hoop stress is calculated as
+ * hoop_stress = tangential^T_i * \sigma_{ij} * tangential_j
+ * where tangential is tangential direction at the current point in the spherical system;
+ * @param center The center point of the spherical system
+ * @param curr_point The point corresponding to the stress (pass in & _q_point[_qp])
+ * @param direction The direction vector in which the scalar stress value is calculated.
+ */
+template <typename T>
+T
+hoopStress(const RankTwoTensorTempl<T> & stress,
+           const Point & center,
+           const Point & curr_point,
+           Point & direction)
+{
+  Point radial = curr_point - center;
+  Real r = radial.norm();
+  radial /= r;
+
+  // Given normal vector N=(n1,n2,n3) and current point C(c1,c2,c3), the tangential plane is then
+  // defined as n1(x-c1) + n2(y-c2) + n3(z-c3)=0. Let us assume n1!=0, the arbitrary point P on this
+  // plane can be taken as P(x,c2+r,c3+r) where r is the radius. The x can be solved as x =
+  // -r(n2+n3)/n1 + c1. The tangential vector PC is given as P-C.
+
+  Point tangential;
+
+  if (!MooseUtils::absoluteFuzzyEqual(radial(0), 0.0))
+  {
+    Real x = curr_point(0) - (radial(1) + radial(2)) * r / radial(0);
+    tangential = Point(x, curr_point(1) + r, curr_point(2) + r) - curr_point;
+  }
+  else if (!MooseUtils::absoluteFuzzyEqual(radial(1), 0.0))
+  {
+    Real y = curr_point(1) - (radial(0) + radial(2)) * r / radial(1);
+    tangential = Point(curr_point(0) + r, y, curr_point(2) + r) - curr_point;
+  }
+  else if (!MooseUtils::absoluteFuzzyEqual(radial(1), 0.0))
+  {
+    Real z = curr_point(2) - (radial(0) + radial(1)) * r / radial(2);
+    tangential = Point(curr_point(0) + r, curr_point(1) + r, z) - curr_point;
+  }
+  else
+    mooseError("In Hoop stress calculation for spherical geometry, the current (quadracture) point "
+               "is likely to be at the center. ");
```

Comment:
You should have no problem getting coverage for this stuff in the unit tests

Reply:
@loganharbour  I am not sure how to do this. Any suggestion?

Reply:
This is a utility; you can make a unit test for it just as we do in https://github.com/idaholab/moose/tree/next/modules/tensor_mechanics/unit/src

Reply:
I agree this is a utility and can be tested as a unit test. However, I have not see any unit tests were added for other scalar quantity calculation. The code has been largely tested in those regression tests. If you do not insistent on this, shall we not add it in this PR?  Maybe we can add all of them in a separate PR in the future.

Reply:
>  Maybe we can add all of them in a separate PR in the future.

This will not happen - but that's alright. @dschwen already gave the ✅

Reply:
:laughing: :laughing: :laughing: